delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchIndexingStreamEntry.scala

```diff
@@ -87,9 +87,19 @@ final case class ElasticSearchIndexingStreamEntry(
     val metaGraph = resource.metadataGraph
     val graph     = if (includeMetadata) predGraph ++ metaGraph else predGraph
     if (sourceAsText)
-      graph.add(nxv.originalSource.iri, resource.source.noSpaces).toCompactedJsonLd(context).map(_.obj.asJson)
+      graph
+        .add(nxv.originalSource.iri, resource.source.noSpaces)
+        .toCompactedJsonLd(context)
+        .map(_.obj.asJson)
+    else if (resource.source.isEmpty())
+      graph
+        .toCompactedJsonLd(context)
+        .map(_.obj.asJson)
     else
-      graph.toCompactedJsonLd(context).map(ld => mergeJsonLd(resource.source, ld.json).removeAllKeys(keywords.context))
+      (graph -- graph.rootTypesGraph)
+        .replaceRootNode(BNode.random) // This is done to get rid of the @id in order to avoid overriding the source @id
```

Comment:
what if the source does not contain an @id?

Reply:
Then there is no @id. It is up to the client to decide that